### ISSUE TITLE
DOS: __brk() missing from stdlib

### ISSUE THREAD

**Issue by darkstar** [open-watcom/open-watcom-v2#276]
The FreeDOS kernel source code does this:
````
#ifdef __WATCOMC__
#include <stdlib.h>
#define brk(newbrk) ((int)__brk((unsigned)(newbrk)))
#endif
```` 

However, in the current ``stdlib.h`` there is no ``__brk()``, only ``sbrk()``, see https://github.com/open-watcom/open-watcom-v2/blob/master/bld/hdr/brk.sp. This is an extract from the generated ``stdlib.h`` file:
````
...
extern  int       __wargc;      /* number of wide cmd line args */
extern  wchar_t **__wargv;      /* vector of wide cmd line args */
#endif

_WCRTLINK extern void _WCNEAR *sbrk( int __increment );

typedef void (*onexit_t)( void );
_WCRTLINK extern onexit_t     onexit( onexit_t __func ); 
...
````

An older version of OpenWatcom (in my case, 1.9) has the ``__brk()`` in that same include file:
````
...
extern  int       __wargc;      /* number of wide cmd line args */
extern  wchar_t **__wargv;      /* vector of wide cmd line args */
#endif

_WCRTLINK extern void _WCNEAR *__brk( unsigned __new_brk_value );
_WCRTLINK extern void _WCNEAR *sbrk( int __increment );

typedef void (*onexit_t)( void );
_WCRTLINK extern onexit_t     onexit( onexit_t __func );
...
````

Is there a specific reason it was removed? Has it moved elsewhere? Can it be reintroduced?

**Comment by darkstar** [comment]
I'm currently trying this diff, it's still compiling:
````
diff --git a/bld/hdr/brk.sp b/bld/hdr/brk.sp
index e9909cf..3827313 100644
--- a/bld/hdr/brk.sp
+++ b/bld/hdr/brk.sp
@@ -5,6 +5,12 @@
 _WCRTLINK extern int          brk( void *endds );
 :endsegment
 :segment QNX
+#else
+:endsegment
+:segment DOS | QNX
+_WCRTLINK extern void _WCNEAR *__brk( unsigned __new_brk_value );
+:endsegment
+:segment QNX
 #endif
 :endsegment
 _WCRTLINK extern void _WCNEAR *sbrk( int __increment );
````

The change was introduced in commit e960cab8accbb0c1171cf497cfe0331fe5e17c00 and it was probably unintended collateral damage that took away ``__brk`` from the DOS builds

Note that this diff is not tested against QNX so if there are any changes required for it then it should not be blindly applied

**Comment by jmalak** [comment]
Thanks for bug report.
Now it is fixed by 578029d02e5c21780552a624ccbe838f716e7b1a commit

**Comment by darkstar** [comment]
Works for me. Thanks for fixing! Closing the issue
